cmd/stackdriver-prometheus-sidecar/main.go

```diff
@@ -490,11 +492,22 @@ type clientFactory struct {
 }
 
 func (f *clientFactory) New() stackdriver.StorageClient {
+	rb, rbcleanup := manual.GenerateAndRegisterManualResolver()
+	defer rbcleanup()
```

Comment:
This will call rbcleanup() when New returns. I believe you need to call it when the client factory is no longer used (which may be never).

Also note that you have a second copy a few lines below.